smarts/core/traffic_history_provider.py

```diff
@@ -18,44 +18,54 @@
 # OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 # THE SOFTWARE.
 from itertools import cycle
-from typing import Set
+from typing import Set, NamedTuple
+import sqlite3
 
 from .controllers import ActionSpaceType
 from .coordinates import BoundingBox, Heading, Pose
 from .provider import Provider, ProviderState
-from .utils.traffic_history_service import Traffic_history_service
 from .vehicle import VEHICLE_CONFIGS, VehicleState
 
 
 class TrafficHistoryProvider(Provider):
     def __init__(self):
         self._is_setup = False
-        self._traffic_history_service = None
         self._map_location_offset = None
-        self.replaced_vehicle_ids = set()
-        self.start_time_offset = 0
+        self._replaced_vehicle_ids = set()
+        self._start_time_offset = 0
+        self._histories_db = None
```

Comment:
Looks like you ordered the instance variables but then added one to put it back out of order. :rofl:

Reply:
hehe, yep!  :)